File: backend/schemas/activities.py
from sqlalchemy import Column, Integer, String, DateTime, Date
from datetime import datetime
from db.db import Base

class Activity(Base):
    __tablename__ = "activities"

    id = Column(Integer, nullable=False, unique=True, primary_key=True)
    couple_id = Column(Integer, nullable=False, foreign_key="couples.id")
    name = Column(String, nullable=False)
    category = Column(String, nullable=False)
    created_at = Column(DateTime, nullable=False, default=datetime.now)

class ActivitySnapshot(Base):
    __tablename__ = "activity_snapshots"
    
    id = Column(Integer, nullable=False, unique=True, primary_key=True)
    visit_id = Column(Integer, nullable=False, foreign_key="visits.id")
    activity_id = Column(Integer, nullable=True, foreign_key="activities.id")
    date = Column(Date, nullable=False)
    name = Column(String, nullable=False)
    category = Column(String, nullable=False)
    order_index = Column(Integer, nullable=False)

File: frontend/src/api/backend/activities.ts
import { authenticatedFetch } from "./apiClient";

export interface CreateActivitySnapshot {
  activity_id: number | null;
  date: string
  name: string
  category: string
  order_index: number
}

export interface ActivitySnapshot {
  id: number;
  visit_id: number;
  activity_id: number | null;
  date: string;
  name: string;
  category: string;
  order_index: number;
}

export async function getActivitySnapshots(visit_id: number): Promise<ActivitySnapshot[]> {
  const response = await authenticatedFetch(`/activities/snapshots/${visit_id}`, {
    method: "GET",
  });
  if (!response.ok) {
    throw new Error("Failed to get activity snapshots");
  }
  const snapshots: ActivitySnapshot[] = await response.json();
  return snapshots;
}